wwwbase/ajax/scramble.php

```diff
@@ -0,0 +1,201 @@
+  $lexem = "";
+  $lexem = randomWordGenerator();  
+  $letters = str_split_unicode($lexem);
+  gen($letters, 0, $results);
+  //print_r($results);
+
+  $wordList = array();
+  $wordList = findWords($results);
+  //print_r($wordList);
+  ajaxEcho($lexem, $wordList);
+
+  //print_r($results);
+
+  //for spliting into substrings a word of 5 letters
+  /*
+  for ($i = 0; $i < 3; $i++) {
+      $subWord = mb_substr($lexem, $i, 3);
+      $lexemArray = str_split_unicode($subWord);
+      $temp = letterPermute($lexemArray);
+      $wordList = array_merge($wordList, $temp);
+    }
+//echo '<pre>'; print_r($wordList); echo '</pre>';
+  for ($i = 0; $i < 2; $i++) {
+    $subWord = mb_substr($lexem, $i, 4);
+    $lexemArray = str_split_unicode($subWord);
+    $temp = letterPermute($lexemArray);
+    $wordList = array_merge($wordList, $temp);
+  }
+//first char + last 2 chars
+  for($i = 0; $i < 2; $i++) {
+    $subWord = mb_substr($lexem, $i, 1); 
+    $subWord .= mb_substr($lexem, 3, 2);
+    $lexemArray = str_split_unicode($subWord);
+    $temp = letterPermute($lexemArray);
+    $wordList = array_merge($wordList, $temp);
+  }
+  //first char + last 3 chars
+  $subWord = mb_substr($lexem, 0, 1);
+  $subWord .= mb_substr($lexem, 2, 3);
+  $lexemArray = str_split_unicode($subWord);
+  $temp = letterPermute($lexemArray);
+  $wordList = array_merge($wordList, $temp);
+  //first 3 chars + last char
+  $subWord = mb_substr($lexem, 0, 3);
+  $subWord .= mb_substr($lexem, 4, 1);
+  $lexemArray = str_split_unicode($subWord);
+  $temp = letterPermute($lexemArray);
+  $wordList = array_merge($wordList, $temp);
+  //permute the whole word.
+  $lexemSplit = $lexem;
+  $lexemArray = str_split_unicode($lexemSplit);
+  $wordList = array_merge($wordList, letterPermute($lexemArray));
+
+  $wordArray = findWords($wordList);
+  ajaxEcho($lexem,$wordArray);
+  */
+
+}
+
+// Permută pozițiile $k ... |$v|-1
+function gen(&$v, $k, &$results) {
+  // Primele $k elemente constituie o soluție (peste o limită minimă)
+  if ($k >= MIN_LENGTH) {
+    $results[] = implode(array_slice($v, 0, $k));
+  }
+
+  // Dacă n-am epuizat vectorul, continui
+  if ($k < count($v)) {
+    // Pe poziția $k pun, pe rând, fiecare din elementele rămase
+    for ($i = $k; $i < count($v); $i++) {
+      $tmp = $v[$k];
+      $v[$k] = $v[$i];
+      $v[$i] = $tmp;
+
+      // Reapelez de la poziția $k + 1
+      gen($v, $k + 1, $results);
+
+      // Apoi pun elementul inițial la loc
+      $tmp = $v[$k];
+      $v[$k] = $v[$i];
+      $v[$i] = $tmp;
+    }
+  }
+  // Astfel, la final, $v este nemodificat
+}
+
+
+function findWords($wordList) {
+  //$wordList1 = array();
+  $wordsFound = array(); //the final words that were found after doing all the possible combinations.
+  $j = 0;
+  for($i = 0 ;$i < count($wordList); $i++) {
+    //$wordList1[$i] = implode($wordList[$i]);
+    $search = Model::factory('Lexem')
+            ->where("formUtf8General", $wordList[$i])
+            ->find_one();
```

Comment:
Am primit uneori de la testeri :), ca uneori nu-mi prinde anumite combinatii, adica dintr-un cuvant care se putea forma sat nu mi-a gasit acea combinatie in spate. Aici nu sunt sigur daca e o problema de generare sau o problema de verificare. Trebuie sa mai ma uit.